WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Posts/PostsCardViewModel.swift

```diff
@@ -121,17 +131,36 @@ private extension PostsCardViewModel {
             ofType: .post,
             with: options,
             for: blog,
-            success: { _ in
+            success: { [weak self] _ in
+                if self?.numberOfPosts == 0 {
+                    self?.showEmptyPostsError()
+                }
+            }, failure: { [weak self] _ in
+                if self?.numberOfPosts == 0 {
+                    self?.showLoadingFailureError()
+                }
+        })
+    }
 
-            }, failure: { (error: Error?) -> () in
+    func showEmptyPostsError() {
+        viewController?.hideLoading()
+        viewController?.showError(message: Strings.noPostsMessage, retry: false)
+    }
 
-        })
+    func showLoadingFailureError() {
+        viewController?.hideLoading()
+        viewController?.showError(message: Strings.loadingFailure, retry: true)
     }
 
     enum Constants {
         static let numberOfPosts = 3
         static let numberOfPostsToSync: NSNumber = 4
     }
+
+    enum Strings {
+        static let noPostsMessage = NSLocalizedString("You don't have any posts", comment: "Displayed when the user views the dashboard posts card but they have no posts")
+        static let loadingFailure: String = NSLocalizedString("Unable to load posts right now.", comment: "Message for when posts fail to load on the dashboard")
```

Comment:
[nit]
```suggestion
        static let loadingFailure = NSLocalizedString("Unable to load posts right now.", comment: "Message for when posts fail to load on the dashboard")
```

Reply:
Done in 44f51bea278c41365cfae9b31bcc36283f26957d